### PR TITLE
Add 1CH mini Smart Switch Model FS-02ZE `_TS000F_xkap8wtb`

### PR DESCRIPTION
**Pull request by arsenyspb**
Original thread [issues/2447](https://github.com/zigpy/zha-device-handlers/issues/2447). Verified on Tuya device model "FS-02ZE". Kudos to @erkr and @TheJulianJES .

Test to be triggered and addressed in-PR itself.

### CODE REVIEW

**Comment by arsenyspb** [comment]
@TheJulianJES @javicalle folks, long time no see! 

Please LGTM - fixes [issue/2447](https://github.com/zigpy/zha-device-handlers/issues/2447) for [active power](https://www.evernote.com/shard/s151/sh/ff9ef2d6-db84-4349-b6fa-12678a247d4a/X2jMDG2xuS4DjP6TPDkXycfejjs-YyMkKt-wpaSq1fWHTzOQXV2zXSKCZw) and [summation delivered](https://www.evernote.com/shard/s151/sh/e9ba60bb-105a-4525-b68e-29153608a03b/ZciKLFJCLq4IzNBXMbJRjcoZ8rt2I-flT6-CDeok7B6iqjhtsWJiG33tkg), but follow my advice, don't buy this cheap stuff IRL please.

File: zhaquirks/tuya/ts000f_switch.py
Comment (MattWestb) [inline zhaquirks/tuya/ts000f_switch.py]: If the device is working OK with the original device type it shall not getting changed then some device dont like it and getting problems in the long run.
Change the device type on the device card or in device config.yaml so the Zigbee type is not making problems.

**Review by MattWestb — CHANGES_REQUESTED** [review]

File: zhaquirks/tuya/ts000f_switch.py
Comment (MattWestb) [inline zhaquirks/tuya/ts000f_switch.py]: The cluster is defined in INIT so importing it and using it in´stead and also in the replacement.

File: zhaquirks/tuya/ts000f_switch.py
Comment (MattWestb) [inline zhaquirks/tuya/ts000f_switch.py:64]: Its no reason deleting the GP EP its better to have it then its possible see its one GP router from the device card.

**Review by MattWestb — CHANGES_REQUESTED** [review]

**Comment by MattWestb** [comment]
### Nice team  work done !!!!

**Comment by erkr** [comment]
Thanks for adopting my custom quirk 🥳

File: zhaquirks/tuya/ts000f_switch.py
Comment (arsenyspb) [reply zhaquirks/tuya/ts000f_switch.py]: Actually as stated in [issue/2447](https://github.com/zigpy/zha-device-handlers/issues/2447) and observed in field the original device type detected is wrong (light), so we set it right here?..

Comment (MattWestb) [reply zhaquirks/tuya/ts000f_switch.py]: Its not nice done from tuya but is the right way do in ZHA for not getting one system that have problems so in the replasment is shall being `DEVICE_TYPE: zha.DeviceType.ON_OFF_LIGHT,`.
And use the method Erik have linked in the issue https://github.com/zigpy/zha-device-handlers/issues/2447#issuecomment-1742132942.

File: zhaquirks/tuya/ts000f_switch.py
Comment (arsenyspb) [reply zhaquirks/tuya/ts000f_switch.py:64]: @MattWestb can you propose a change here please?.. I'm sorry it's very hard to interpret "no reason deleting GP EP", maybe it's something obvious but I'm missing it completely... let me learn from you?

File: zhaquirks/tuya/ts000f_switch.py
Comment (arsenyspb) [reply zhaquirks/tuya/ts000f_switch.py]: @MattWestb could you share with me the code proposal here? I'd like to learn and pick up your skills but from just comment it's hard to interpret for me unfortunately (...lack of background knowledge I guess, rookie here).

File: zhaquirks/tuya/ts000f_switch.py
Comment (arsenyspb) [reply zhaquirks/tuya/ts000f_switch.py]: @MattWestb patched in [4009eca](https://github.com/zigpy/zha-device-handlers/pull/2613/commits/4009eca17f801f5de6e0228dc15d31b4823bc854).

**Comment by arsenyspb** [comment]
> ### Nice team work done !!!!

Look forward to your best-practice knowledge sharing and coaching - please suggest the solution further for your comments... 🙏 let's resolve them

File: zhaquirks/tuya/ts000f_switch.py
Comment (MattWestb) [reply zhaquirks/tuya/ts000f_switch.py:64]: Im sorry im not maintainer but @TheJulianJES can if hi have time doing that.

File: zhaquirks/tuya/ts000f_switch.py
Comment (MattWestb) [reply zhaquirks/tuya/ts000f_switch.py]: Put it in the import like the other tuya cluster.
```
from zhaquirks.tuya import (
    TuyaZBE000Cluster,
    TuyaZBElectricalMeasurement,
    TuyaZBMeteringCluster,
    TuyaZBOnOffAttributeCluster,
)
```
and then using it like like the other is being done (dont braking the sorting in the import).
The name shall being `TuyaZB1888Cluster` if i have reading it OK.

Comment (arsenyspb) [reply zhaquirks/tuya/ts000f_switch.py]: @MattWestb please take my feedback without any negativity or arrogance, but it's **really** hard to interpret your written English... 🙏  Frankly, I don't see why we need to sophisticate it. Considering my past experience e.g. in [pull/2252](https://github.com/zigpy/zha-device-handlers/pull/2252), and ability for community contributors to superseed this initial quirk submission, as well as following [small change code review best-practices](https://abseil.io/resources/swe-book/html/ch09.html#write_small_changes), I believe we could've just went past this.

**However**, - @erkr - do you have a second opinion on this? Can you second the reviewer's suggestion and propose (...and most importantly cross- test on your end, since there are only two of us with these cheap devices) the suggestion of:

```
from zhaquirks.tuya import (
    TuyaZB1888Cluster
    TuyaZBE000Cluster,
    TuyaZBElectricalMeasurement,
    TuyaZBMeteringCluster,
    TuyaZBOnOffAttributeCluster,
)
```
...as well as change from `0x1888,` to `TuyaZB1888Cluster.cluster_id,`?

File: zhaquirks/tuya/ts000f_switch.py
Comment (arsenyspb) [reply zhaquirks/tuya/ts000f_switch.py:64]: @TheJulianJES apologies for interrupting you - I'm really not following the suggestion here, can you please invest a tiny bit of your precious time and share knowledge with @erkr and myself: what exactly is @MattWestb proposing?..

Comment (MattWestb) [reply zhaquirks/tuya/ts000f_switch.py:64]: Only adding the second endpoint in the replacement like normal Zigbee devices is having like this:
https://github.com/zigpy/zha-device-handlers/blob/74b9f043a996da6baca722d1e191d842a31ae198/zhaquirks/ikea/tradfriplug.py#L156-L166

I have taking some lines from the first ep with so you see how its being done.

Comment (arsenyspb) [reply zhaquirks/tuya/ts000f_switch.py:64]: Thank you for pointing this snippet! Now I fully understand... I wasn't even aware about Zigbee Green Power (ZGP) before, just read up about the routing and sinking the metrics, now I understand why it's important to pass it on 🙏  Addressed this, resolving this review.

File: zhaquirks/tuya/ts000f_switch.py
Comment (arsenyspb) [reply zhaquirks/tuya/ts000f_switch.py]: I've tested it and it works, resolving.

**Comment by arsenyspb** [comment]
@MattWestb please cross-check once again, I think it's now fully aligned to community best practices and contribution guidelines... Thanks!

**Review by MattWestb — APPROVED** [review]
For me its looks very OK but the maintainer shall looking more in deep of the cod parts that is out of my knowledge and i hope they is merging it.
## Great work done !!!

File: zhaquirks/tuya/ts000f_switch.py
Comment (TheJulianJES) [inline zhaquirks/tuya/ts000f_switch.py:91]: I think this is what @MattWestb mentioned.
The ZGP endpoint is "deleted" (ZHA won't see it) when it's not re-defined in the `replacement`.

Done through the GitHub UI (so no validation), but I think this should be the change needed here:
```suggestion
            242: {
                PROFILE_ID: zgp.PROFILE_ID,
                DEVICE_TYPE: zgp.DeviceType.PROXY_BASIC,
                INPUT_CLUSTERS: [],
                OUTPUT_CLUSTERS: [GreenPowerProxy.cluster_id],
            },
```

File: zhaquirks/tuya/ts000f_switch.py
Comment (TheJulianJES) [inline zhaquirks/tuya/ts000f_switch.py]: I see why you'd want to do this, but this is currently the case for all Tuya "light switches" and I'm not sure a link shortener link is a good idea. So, let's remove the comment here.
```suggestion
```

**Review by TheJulianJES — COMMENTED** [review]
Thanks for the PR!

Two small changes needed:
- "ZigbeeGreenPower" endpoint `242` needs to be added to the replacement. The section can just be copied from the `signature`
- The comment about overriding the device type shouldn't be in the quirk itself.

**Review by javicalle — APPROVED** [review]
Maybe the `Tuya_1G_Wall_Switch_Metering` looks a bite long overdescriptive to ne, but just an aesthetical matter.

LGTM

**Comment by erkr** [comment]
@arsenyspb 
Thanks for organising this pull request. Best Eric

**Review by TheJulianJES — APPROVED** [review]
Thanks again! Also LGTM

**Pull request merged**
